### DISCUSSION TITLE
Linux Volume Questions

### DISCUSSION THREAD

**Discussion by advice2020** [mpv-player/mpv#14095] [Q&A]
Linux Debian based distro

Hopefully this is okay, but I have some questions about Linux / MPV volume controls.
Most of the questions have to do with using desktop environment system volume for volume controls instead of MPV internal volume controls.

1.) So far I have been able to figure out using the “run” command in MPV which enables the access to the ALSA mixer volume control
`run amixer set Master 5%+`
`run amixer set Master 5%-`
`run amixer set Master toggle`
This works but was just wondering if anyone knows of a better method to achieve this in MPV, so far this seems like the best approach.

2.) Not sure if this is even possible but I was wondering if there is any way of linking the actual MPV volume bar OSD to the ALSA mixer volume value?
So for example, if the ALSA mixer volume value (Desktop environment volume value) is set to 60%, then if you open up MPV the Volume Bar OSD shows 60%

-(Not as important because I might remove from my OSC, but) If the above is possible, is it possible for this to also be linked to the OSC volume speaker icon?  Especially the “mute” icon that shows up as speaker with X.

-The desktop environment that I am using views its own Volume OSD as a notification, so this is what currently pops up in the upper right corner on top of MPV (fullscreen).  This is okay, but I like the MPV volume bar better.  Shows up in center of screen and is much easier to see.
If MPV volume can be linked the only downside is that then it would show 2 volume OSDs at the same time so I would have to turn off the DE volume notification OSD, which would then cause this volume OSD to not show up in any applications, because I do not think it would be possible to only turn off volume notification for MPV only, unless someone can think of a way to achieve this.

Thank you to anyone reads this and for any info shared.

**Comment by advice2020** [comment]
Not sure if these are similar to what I am looking for but 
-I have seen others mention in the past `softvol=no` which is said to no longer work.  
-Also I have seen people mention the solution for this could also be using `ao-volume` in place of volume keys but this did not achieve what I am looking for.

**Comment by kasper93** [comment]
Have you seen `ao-volume` property? https://mpv.io/manual/master/#command-interface-ao-volume

**Comment by advice2020** [comment]
@kasper93 
I just said that! :)
(I posted my reply right after my main post, which was probably not visible to you in time before you made your post)
First off thank so much for your quick reply.
I apologize Linux audio confuses me a bit.
To answer your question, yes I have seen others mention `ao-volume` which I did try.
This did not link to my ALSA, the only thing I noticed was that instead of saying "VOLUME" at top of volume OSD, it says something like "AOVOLUME" so that probably means like the manual says maybe it is using "PulseAudio" which my system does run.
What confuses me is that I thought that my DE speaker icon although is "PulseAudio" that it updates with ALSA mixer.
For example if I run AlsaMixer from terminal and I scroll on the mixer fader my desktop environment "PulseAudio" speaker icon updates with it, so I was just thinking I would try to link MPV to the ALSA mixer volume itself.
Manual says "while with PulseAudio this controls per-application volume."
So I am a bit confused, not sure how to point MPV to ALSA so that it uses global volume as opposed to "PulseAudio" per application that it appears to be doing currently.
and to make things even more confusing Pipewire is also on my system as well :)

Thanks Again

**Comment by advice2020** [comment]
Still interested if anyone has any more ideas.
Thank You

**Comment by advice2020** [comment]
Not sure if this was there before but I noticed some more information in the manual about this.
It says that `ao-volume` achieves this when your audio driver is set to ALSA but if using PulseAudio `ao-volume` will adjust the volume per application, which does not really make sense to me because isn't this already what the MPV volume OSD is doing anyways?  It is just adjusting its own volume.
I am using PulseAudio so that is why I mentioned I am getting the results that I am getting.
My MPV volume OSD by default was showing "VOLUME" text at the top screen but after setting `ao-volume` the text would read "AO-VOLUME" but it was only adjusting MPV the same as it did before this `ao-volume` setting.

Does anyone know if there is any way of getting PulseAudio to perform as how ALSA is said to work with `ao-volume`?  Without actually having to change driver to ALSA.

Confusing, because if I open up my alsa-mixer GUI (vertical volume fader) and then I adjust the volume on my PulseAudio speaker icon in panel, the alsa-mixer fader moves up and down with it.  So if PA adjusts ALSA then why wouldn't MPV adjusting PA then adjust ALSA?

Anyways hopefully some of this makes some sense to someone, if anyone has any information they could share please let me know.
Thank you to anyone who reads this

**Comment by advice2020** [comment]
Not sure if this helps but turns out my system is actually using Pipewire that is using PulseAudio.
I thought I read something about Pipewire and MPV but cannot remember.
I figured I would post this just in case I cannot find anything about this.
Still very interested if anyone has any ideas of how to achieve this?
Looking to have MPV volume OSD to be linked / use system audio instead of its own internal MPV program based volume.
Thank You for any info shared

**Comment by advice2020** [comment]
I still have these hotkeys set in my input.conf file which I believe are correct.
```
9 add ao-volume -2
0 add ao-volume 2
m cycle ao-mute
```
I saw these in manual
```
--pipewire-volume-mode=<channel|global>
Specify if the ao-volume property should apply to the channel volumes or the global volume. By default the channel volumes are used.
```
I tried adding the following to my mpv.conf file but did not notice my system volume meter changing.
`pipewire-volume-mode=global`

I also saw the following in the manual, I tried some things out but not sure if this is what I am looking for, wasn't sure what to put in for the <device> name, is this looking for hardware or could this be linked to Pipewire-pulse (PulseAudio) in any way, to link it to control the ALSA mixer fader?
```
--alsa-mixer-device=<device>
Set the mixer device used with ao-volume (default: default).
```
Anyways if anyone knows of anything else I could try, please let me know.
In general has anyone ever gotten this to work on their machine?
Thank You
